src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -1384,6 +1399,9 @@ void DatabaseCatalog::dropTableFinally(const TableMarkedAsDropped & table)
     LOG_INFO(log, "Removing metadata {} of dropped table {}", table.metadata_path, table.table_id.getNameForLogs());
     fs::remove(fs::path(table.metadata_path));
 
+    LOG_DEBUG(log, "Try remove permanently flag for detached table {}", table.table_id.getNameForLogs());
+    removeDetachedPermanentlyFlag(table);
```

Comment:
It has to be removed when enqueueing the table for drop

Reply:
If we have a restart Clickhouse between enqueueing the table for drop and executing drop we will attach table. It seems to me that It will undesirable behavior.